miner/miner.go

```diff
@@ -119,20 +139,38 @@ func (miner *Miner) update() {
 	}
 }
 
+// Start starts the miner mining, unless it has been paused by the downloader
+// during sync, in which case it will start mining once the sync has completed.
 func (miner *Miner) Start(coinbase common.Address) {
-	atomic.StoreInt32(&miner.shouldStart, 1)
-	miner.SetEtherbase(coinbase)
-
-	if atomic.LoadInt32(&miner.canStart) == 0 {
+	miner.startStopMutex.Lock()
+	defer miner.startStopMutex.Unlock()
+	miner.shouldStart = true
+	if !miner.canStart {
```

Comment:
It seems that if this path is hit, the `miner.SetEtherbase` doen't happen, and later on, it will start mining to the `0`-address.